### PR TITLE
Support for multi-file transactions in file repo

### PR DESCRIPTION
**Pull request by mkot02**
Added support for adding and removing many file content units from repository using one command.
This feature was already implemented in pulp-file plugin.

fixes #8120

### CODE REVIEW

File: pulpcore/cli/file/repository.py [hunk 1]
```diff
@@ -202,27 +202,44 @@ def remove(
...
+    option_value: str,
+) -> List[Dict[str, str]]:
+    """Load JSON from input string or from file if string starts with @."""
+    json_object: List[Dict[str, str]]
+    json_string: Union[str, bytes]
+
+    if option_value.startswith("@"):
+        json_file = option_value[1:]
+        try:
+            with open(json_file, "r") as fp:
```
Comment (mdellweg) [inline pulpcore/cli/file/repository.py:215]:
```suggestion
            with open(json_file, "rb") as fp:
```

File: pulpcore/cli/file/repository.py [hunk 2]
```diff
@@ -202,27 +202,44 @@ def remove(
...
-    default="[]",
-    help="JSON string with list of objects with 'sha256' and 'relative_path' keys",
-)
-@click.option(
-    "--remove-content",
-    default="[]",
-    help="JSON string with list of objects with 'sha256' and 'relative_path' keys",
-)
+@click.option("--add-content", default="[]")
+@click.option("--remove-content", default="[]")
```
Comment (mdellweg) [inline pulpcore/cli/file/repository.py:234]: Please keep the help text.

File: pulpcore/cli/file/repository.py [hunk 3]
```diff
@@ -202,27 +202,44 @@ def remove(
     )
 
 
+def _load_json_from_option(
```
Comment (mdellweg) [inline pulpcore/cli/file/repository.py:205]: You can turn this function into an option callback. Then you will just have the python objects in the command body.

https://click.palletsprojects.com/en/7.x/options/#callbacks-and-eager-options

File: pulpcore/cli/file/repository.py (hunk 1 above)
Comment (mkot02) [reply pulpcore/cli/file/repository.py]: Done

File: pulpcore/cli/file/repository.py (hunk 2 above)
Comment (mkot02) [reply pulpcore/cli/file/repository.py]: Restored and improved help text

File: pulpcore/cli/file/repository.py [hunk 4]
```diff
@@ -212,34 +214,48 @@ def _load_json_from_option(
...
+    default="[]",
+    callback=_load_json_from_option,
+    is_eager=True,
+    expose_value=True,
+    help="JSON string with a list of objects with the keys 'sha256', 'relative_path'. The argument prefixed with the '@' can be the path to a JSON file with a list of objects.",
+)
+@click.option(
+    "--remove-content",
+    default="[]",
+    callback=_load_json_from_option,
```
Comment (mdellweg) [inline pulpcore/cli/file/repository.py:248]: Now that looks neat. ;)

**Review by mdellweg — APPROVED** [review]

**Review by daviddavis — APPROVED** [review]
This is great. Thank you for the contribution! :tada:

**Pull request merged**